### ISSUE TITLE
required_if doesn't work as expected

### ISSUE THREAD

**Issue by risanto** [skaterdav85/validatorjs#434]
So, I set up these rules:
```
channel: 'required|numeric',
url: 'required_if:channel,1|url_validation',
```
And it's supposed to check for url when the channel is 1, but the problem is when I send this payload it passes just fine and doesn't trigger required or the subsequent url_validation (a custom validation).
```
"channel": 1,
"url": null,
```

**Comment by min-emcloud** [comment]
```
const Validator = require('validatorjs')

const data = {
  name: 'John',
  hasAllergy: true
};

const rules = {
  name: 'required',
  hasAllergy:  'required|boolean',
  allergicTo: 'required_if:hasAllergy,true'
}

const validation = new Validator(data, rules)
console.log('validation result', validation.passes()) // expected `false` but shows `true`
```

**Comment by hellohasan** [comment]
@risanto did you find the solution?

**Comment by AlessandroStaffolani** [comment]
As shown in this closed issue #252. You can solve it using the array/object notation. Here is a working example:

```javascript
const Validator = require('validatorjs');

const data = {
  name: 'John',
  hasAllergy: true,
};

const rules = {
  name: ['required'],
  hasAllergy: ['required', 'boolean'],
  allergicTo: [{ required_if: ['hasAllergy', true] }],
};

const validation = new Validator(data, rules);
console.log('validation result', validation.passes()); // expected `false` and shows `false`
```
